src/global-table.ts

```diff
@@ -0,0 +1,68 @@
+import { IResource, RemovalPolicy, Resource, Stack } from 'aws-cdk-lib';
+import * as ddb from 'aws-cdk-lib/aws-dynamodb';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+import * as perms from './perms';
+
+export interface IGlobalTable extends IResource {
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+  grantReadData(grantee: iam.IGrantable): iam.Grant;
+}
+
+abstract class GlobalTableBase extends Resource implements IGlobalTable {
+  public abstract readonly tableArn: string;
+  protected readonly regionalArns = new Array<string>();
+  public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [
+        this.tableArn,
+      ],
+      scope: this,
+    });
+  }
+
+  public grantReadData(identity: iam.IGrantable): iam.Grant {
+    // usually, we would grant some service-specific permissions here,
+    // but since this is just an example, let's use S3
```

Comment:
Looks like you copied this from S3.  Remove S3 reference.

Reply:
this is a mixture from [dynamodb repo](https://github.com/aws/aws-cdk/blob/de692900d7d8dbe53bef0dd82ef37f45dc1c90e9/packages/%40aws-cdk/aws-dynamodb/lib/table.ts#L696) and [example construct](https://github.com/aws/aws-cdk/blob/50ce07f75e61ee1a72193f85998d7535a274c8a7/packages/%40aws-cdk/example-construct-library/lib/example-resource.ts#L184). Will clean the comments later!